src/engine/runtime.js

```diff
@@ -699,10 +705,11 @@ class Runtime extends EventEmitter {
             this._refreshTargets = false;
         }
 
-        // @todo(vm#570) only emit if monitors has changed since last time.
-        this.emit(Runtime.MONITORS_UPDATE,
-            Object.keys(this._monitorState).map(key => this._monitorState[key])
-        );
+        if (!this._prevMonitorState.equals(this._monitorState)) {
+            const monitorStateObj = this._monitorState.toJS();
+            this.emit(Runtime.MONITORS_UPDATE, Object.keys(monitorStateObj).map(key => monitorStateObj[key]));
```

Comment:
I think you could simplify this with `this._monitorState.map(value => value).toJS()`. But I think we should think about passing the immutable object over to the GUI, since `toJS` incurs a performance penalty, so we should avoid it if possible.  If we need to cast to JS on the GUI side then we can do it there, but if we don't need to, all the better.